parquet-avro/src/test/java/org/apache/parquet/avro/TestArrayCompatibility.java

```diff
@@ -653,8 +654,8 @@ public void testAvroCompatRequiredGroupInListWithSchema() throws Exception {
     Path test = writeDirect(
         "message AvroCompatRequiredGroupInListWithSchema {" +
             "  optional group locations (LIST) {" +
-            "    repeated group array {" +
-            "      optional group element {" +
+            "    repeated group bag {" +
+            "      optional group array_element {" +
```

Comment:
@liancheng, this is testing something slightly different. The backward compatibility rules are strictly implemented in AvroSchemaConverter and are applied when trying to decide what the data within a Parquet schema represents. On the other hand, when an Avro schema is present in the file we know exactly what was represented.

So this is testing that we can override the backward-compatibility rules using an Avro schema when the converters are initialized. In this case, although the schema appears to be a 2-level list, we're requesting that it be interpreted as a 3-level list. I think that this test is correct.